FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureTableClientFactory.cs

```diff
@@ -0,0 +1,63 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Azure.Core;
+using Azure.Data.Tables;
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.Common.Authentication;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Jobs
+{
+    public class AzureTableClientFactory : IAzureTableClientFactory
+    {
+        private const string StorageEmulatorConnectionString = "UseDevelopmentStorage=true";
+        private readonly TokenCredential _tokenCredential;
+
+        private readonly string _tableUrl;
+
+        public AzureTableClientFactory(
+            IOptions<JobConfiguration> config,
+            ITokenCredentialProvider credentialProvider)
+        {
+            EnsureArg.IsNotNull(config, nameof(config));
+            EnsureArg.IsNotNullOrWhiteSpace(config.Value.TableUrl, nameof(config.Value.TableUrl));
+
+            _tableUrl = config.Value.TableUrl;
+
+            EnsureArg.IsNotNull(credentialProvider, nameof(credentialProvider));
+            _tokenCredential = credentialProvider.GetCredential(TokenCredentialTypes.Internal);
+        }
+
+        public AzureTableClientFactory(
+            ITokenCredentialProvider credentialProvider)
+        {
+            EnsureArg.IsNotNull(credentialProvider, nameof(credentialProvider));
+
+            _tableUrl = StorageEmulatorConnectionString;
```

Comment:
Question. Is this constructor only be used in PaaS environment? and always use emulator connection string?

Reply:
This constructor only is used for the local tests, so the emulator connection string is used. Use another constructor for the product environment.

Reply:
Per our offline discussion, we don't provide a connection string interface at this pr.  And @sowu880 will add it later.